.github/workflows/ci.yml

```diff
@@ -14,25 +15,30 @@ jobs:
 
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Use Node.js 16.x
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
           node-version: 16.x
       - run: yarn --immutable --ignore-engines
-
-      - name: "Check Code Format"
-        run: yarn prettier:check
-
-      - name: "Lint Code"
-        run: yarn lint
+      - name: Build Code
+        run: scripts/build
+      - name: Upload build artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: build-artifacts
+          path: |
+            ./build
+            ./build-turbo
+            ./src/assets
```

Comment:
I've found previously that uploading and then downloading artifacts like this can be slower than just rebuilding everything in the first place because GitHub Action's caches are so slow. Since only one thing depends on this (`test`), I'm not sure why it's necessary? Perhaps this is because you planned to add the `lint` task as well?

Reply:
@benjie Hey! I hope you're feeling better. This change was made because the current webpack is hard to upgrade and won't run on versions of Node greater than 16. By locking it in to run on the latest version that it's compatible with unlocks the potential to run 18/20 in CI without replacing Webpack (which I did attempt and it was very messy, for me at least).

Reply:
Ah yes, I remember you saying about this before. Sorry for forgetting!